src/pages/webinar/[webinarSlug]/index.tsx

```diff
@@ -0,0 +1,252 @@
+import React, { useRef, useEffect } from 'react';
+import { toPng } from 'html-to-image';
+import {
+  FlexContainer,
+  Section,
+  Text,
+  Image,
+  CertificateContent,
+} from '@/components';
+import { WebinarPageProps } from '@/interfaces';
+import { getWebinarPageProps } from '@/utils';
+import { useApi, useUser } from '@/hooks';
+import { useState } from 'react';
+import { routes } from '@/constant';
+import { WebinarHeroContainer } from '@/components';
+
+const WebinarPage = ({
+  webinarId,
+  hostName,
+  hostImageUrl,
+  hostRole,
+  dateAndTime,
+}: WebinarPageProps) => {
+  const { user, isAuth } = useUser();
+  const [userName, setUserName] = useState('');
+  const [userEmail, setUserEmail] = useState('');
+  const [showCertificate, toggleCertificate] = useState(false);
+  const [showRegistrationErrMsg, toggleRegistrationErrMsg] = useState(false);
+
+  const certificateRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    if (user) {
+      setUserName(user.name || '');
+      setUserEmail(user.email || '');
+    }
+  }, [user]);
+
+  const handleDownload = async () => {
+    if (certificateRef.current) {
+      try {
+        const dataUrl = await toPng(certificateRef.current, { quality: 1 });
+        const link = document.createElement('a');
+        link.href = dataUrl;
+        link.download = `${userName}-webinar-certificate.png`;
+        link.click();
+      } catch (error) {
+        console.error('Error generatingcertificate image:', error);
+      }
+    }
+  };
+
+  const { makeRequest } = useApi(
+    'webinar',
+    {
+      url: `${routes.api.webinar}/${webinarId}`,
+    },
+    { enabled: !!user?.email }
+  );
+
+  const onGenerateCertificate = async () => {
+    try {
+      const response = await makeRequest({
+        url: `${routes.api.webinar}/${webinarId}?email=${user?.email}`,
+      });
+
+      if (!response?.status) {
+        throw new Error(`Error occured`);
+      }
+
+      if (response?.data?.isRegistered) {
+        toggleCertificate(true);
+        toggleRegistrationErrMsg(false);
+      } else {
+        toggleCertificate(false);
+        toggleRegistrationErrMsg(true);
+      }
+    } catch (error) {
+      console.log('Error while generating certificate: ', error);
+    }
+  };
+
+  const webinarDate = new Date(dateAndTime);
+  const currentDate = new Date();
+  const isWebinarOver = currentDate >= webinarDate;
```

Comment:
We can create a function and i think we already have a function that tells if a date is earlier or later to a date.